assets/common/shaders/ppx/Sampling.hlsli

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SAMPLING_HLSLI
+#define SAMPLING_HLSLI
+
+// https://www.reedbeta.com/blog/hash-functions-for-gpu-rendering/
+uint PCGHash(inout uint rngState)
+{
+    uint state = rngState;
+    rngState = rngState * 747796405u + 2891336453u;
+    uint word = ((state >> ((state >> 28u) + 4u)) ^ state) * 277803737u;
+    return (word >> 22u) ^ word;
+}
+
+float Random01(inout uint input)
+{
+    return PCGHash(input) / 4294967296.0;
+}
+
+float2 Hammersley(uint i, uint N)
+{
+    uint bits = (i << 16u) | (i >> 16u);
+    bits      = ((bits & 0x55555555u) << 1u) | ((bits & 0xAAAAAAAAu) >> 1u);
+    bits      = ((bits & 0x33333333u) << 2u) | ((bits & 0xCCCCCCCCu) >> 2u);
+    bits      = ((bits & 0x0F0F0F0Fu) << 4u) | ((bits & 0xF0F0F0F0u) >> 4u);
+    bits      = ((bits & 0x00FF00FFu) << 8u) | ((bits & 0xFF00FF00u) >> 8u);
+    float rdi = float(bits) * 2.3283064365386963e-10f;
```

Comment:
nit: the .f suffix can be omitted, HLSL considers `1.0` to be a float by default, and requires `1.0L` for doubles.

Reply:
Removed 'f'.